src/fireEvent.js

```diff
@@ -23,7 +23,10 @@ const isEventEnabled = (
 ) => {
   if (isTextInput(element)) return element?.props.editable !== false;
 
-  return touchResponder?.props.onStartShouldSetResponder?.() !== false;
+  return !(
+    touchResponder?.props.onStartShouldSetResponder?.() === false ||
+    touchResponder?.props.onMoveShouldSetResponder?.() === false
+  );
```

Comment:
```suggestion
  const isAcceptingTouchStart = touchResponder?.props.onStartShouldSetResponder?.() !== false;
  const isAcceptingTouchMove = touchResponder?.props.onMoveShouldSetResponder?.() !== false;
  
  return isAcceptingTouchStart || isAcceptingTouchMove;
```

To clarify the reason for the original `onStartShouldSetResponder() !== false` contion, it is meant to mean:
* `onStartShouldSetResponder() === true` view explicitly wants to accept touch events
* `onStartShouldSetResponder() === false` view explicitly wants to reject touch events
* `onStartShouldSetResponder == undefined` view did not explicitly declare accepting touch events, which in real RN runtime env would probably prevent accepting touch events BUT in order to simulate events on various components in Jest runtime env we assume that missing value means accepting the touch event.

Having that it mind, I wonder is it possible that one of  `onStartShouldSetResponder` & `onMoveShouldSetResponder` returns explicit `true`, while the other return explicit `false`?

Reply:
Yeah, Start can return false and Move return true, so I had to add additional checks. Now it seems to work well.

@thymikee yeah, it really needed tests :D